### PR TITLE
Improve handling of 422 already submitted

### PR DESCRIPTION
**Pull request by mycp98**

### CODE REVIEW

File: app/uk/gov/hmrc/plasticpackagingtaxreturns/util/EdgeOfSystem.scala [hunk 1]
```diff
@@ -17,10 +17,10 @@
...
 
 import java.security.MessageDigest
-import java.util.Base64
+import java.util.{Base64, UUID}
 
 class EdgeOfSystem {
   def getMessageDigestSingleton: MessageDigest = MessageDigest.getInstance("SHA-256")
   def createEncoder: Base64.Encoder = Base64.getEncoder
-
+  def createUuid: UUID = UUID.randomUUID()
```
Comment (liampan) [inline app/uk/gov/hmrc/plasticpackagingtaxreturns/util/EdgeOfSystem.scala:25]: nice

**Review by liampan — CHANGES_REQUESTED** [review]
Looks good, remove purplePrinter and approved :)

File: test/uk/gov/hmrc/plasticpackagingtaxreturns/connectors/ReturnsConnectorSpec.scala [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+class ReturnsConnectorSpec extends PlaySpec with BeforeAndAfterEach with Logging {
+
+  private val httpClient = mock[HttpClient]
+  private val appConfig = mock[AppConfig]
+  private val metrics = mock[Metrics](RETURNS_DEEP_STUBS)
+  private val auditConnector = mock[AuditConnector]
+  private val edgeOfSystem = mock[EdgeOfSystem]
+  private val headerCarrier = mock[HeaderCarrier]
+  private val timer = mock[Timer.Context]
+  private val shhLogger = mock[Logger]
```
[the remark this answers is not in the document]
Comment (lmcewanhmrc) [reply test/uk/gov/hmrc/plasticpackagingtaxreturns/connectors/ReturnsConnectorSpec.scala:51]: noisy, but yeah needs a better name

File: app/uk/gov/hmrc/plasticpackagingtaxreturns/util/PurplePrint.scala [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.plasticpackagingtaxreturns.util
+
+object PurplePrint {
```
[the remark this answers is not in the document]
Comment (lmcewanhmrc) [reply app/uk/gov/hmrc/plasticpackagingtaxreturns/util/PurplePrint.scala:19]: would like to keep it as an easy debug tool - where best to put it? any ideas?

[the remark this answers is not in the document]
Comment (liampan) [reply app/uk/gov/hmrc/plasticpackagingtaxreturns/util/PurplePrint.scala:19]: 🤔

**Review by liampan — APPROVED** [review]

**Review by liampan — APPROVED** [review]

**Pull request merged**